File: LeetCode/02181. Merge Nodes in Between Zeros/02181.c
struct ListNode* mergeNodes(struct ListNode* head){
    struct ListNode* node, *res_index;
    struct ListNode* pre_head = malloc(sizeof(struct ListNode));
    pre_head->next = head;
    int sum;
    
    for (sum=0, res_index=pre_head, node=head->next; node!=NULL; node=node->next) {
        if (node->val != 0) {
            sum+=node->val;
        } else {
            if (sum != 0) {
                res_index=res_index->next;
                res_index->val=sum;
            }
            sum=0;
        }
    }
    res_index->next=NULL;
    
    return head;
}

File: LeetCode/02181. Merge Nodes in Between Zeros/02181.go
// https://leetcode.com/problems/merge-nodes-in-between-zeros

func mergeNodes(head *ListNode) *ListNode {
    last := &ListNode{}

    for prev, cur := head, head.Next; cur != nil; cur = cur.Next {
        if cur.Val != 0 {
            prev.Val += cur.Val
        } else {
            last = prev
            prev.Next = cur
            prev = cur
        }
    }

    last.Next = nil
    return head
}
